### PR TITLE
Test case for tf-idf vectorizer & include licensing info

### PR DESCRIPTION
**Pull request by x65han**
- Add boilerplate at top of `vectorizer.py`
- Add test case for tfidf vectorizer
- Add doc_str for tfidf_vectorizer's class and public method

### CODE REVIEW

File: tests/test_indexutils.py [hunk 1]
```diff
@@ -41,8 +44,21 @@ def setUp(self):
...
+        vectorizer = TfidfVectorizer(self.index_path, min_df=5)
+        train_docs = ['CACM-0239', 'CACM-0440', 'CACM-3168', 'CACM-3169']
+        train_labels = [1, 1, 0, 0]
+        test_docs = ['CACM-0634', 'CACM-3134']
+        train_vectors = vectorizer.get_vectors(train_docs)
+        test_vectors = vectorizer.get_vectors(test_docs)
+        clf = MultinomialNB()
+        clf.fit(train_vectors, train_labels)
+        pred = clf.predict(test_vectors)
+        self.assertListEqual([1, 0], list(pred))
```
Comment (lintool) [inline tests/test_indexutils.py:61]: Can you add an assertion of the actual classifier score?
This will defend against underlying sklearn changes.

File: tests/test_indexutils.py [hunk 2]
```diff
@@ -66,7 +82,8 @@ def test_analyze(self):
         self.assertEqual(' '.join(self.index_utils.analyze('rapid retrieval, space economy')),
                          'rapid retriev space economi')
         tokenizer = pyanalysis.get_lucene_analyzer(stemming=False)
-        self.assertEqual(' '.join(self.index_utils.analyze('retrieval', analyzer=tokenizer)), 'retrieval')
```
Comment (lintool) [inline tests/test_indexutils.py:69]: Are these lines over 100 chars? Unnecessary changes?

Comment (lintool) [reply tests/test_indexutils.py:69]: And below as well...

Comment (x65han) [reply tests/test_indexutils.py:69]: Yea, these are 107 chars. I changed my pylint line length = 100.

File: tests/test_indexutils.py (hunk 1 above)
Comment (x65han) [reply tests/test_indexutils.py]: done

File: tests/test_indexutils.py [hunk 3]
```diff
@@ -41,8 +44,22 @@ def setUp(self):
...
+    def test_tfidf_vectorizer(self):
+        vectorizer = TfidfVectorizer(self.index_path, min_df=5)
+        train_docs = ['CACM-0239', 'CACM-0440', 'CACM-3168', 'CACM-3169']
+        train_labels = [1, 1, 0, 0]
+        test_docs = ['CACM-0634', 'CACM-3134']
+        train_vectors = vectorizer.get_vectors(train_docs)
+        test_vectors = vectorizer.get_vectors(test_docs)
+        clf = MultinomialNB()
+        clf.fit(train_vectors, train_labels)
+        pred = clf.predict(test_vectors)
```
Comment (lintool) [inline tests/test_indexutils.py:60]: I meant, let's assert on the `pred` prediction scores?
